### ISSUE TITLE
from my_module import *

### ISSUE THREAD

**Issue by tomamic** [brython-dev/brython#682]
For my own module, the star import does not work anymore.
Is it an issue for anybody else?

Many thanks for helping!

**Comment by jonathanverner** [comment]
It works for me, here... Could you be more specific? What is the exact layout of your module? Which version of Brython are you using... What is the error?

**Comment by tomamic** [comment]
My module is this one: https://github.com/tomamic/fondinfo/blob/master/bounce/game2d.py
I use it this way: https://github.com/tomamic/fondinfo/blob/master/bounce/bounce_g2d.py

It works if I import names explicitly, one by one.
The "star import" still works fine, with brython_dist.js of 2017-09-04 (maybe also with some later version):

https://github.com/brython-dev/brython/blob/3dd0b0e648e75100d0c4806c39fd18edbea927b6/www/src/brython_dist.js?raw=true

This is the error:

_b_.NameError http://127.0.0.1:8000/brython_dist.js%20line%206432%20%3E%20eval:73:375
$B.$global_search http://127.0.0.1:8000/brython_dist.js:4871:7
<anonymous> http://127.0.0.1:8000/brython_dist.js%20line%204174%20%3E%20eval:217:13
run_script http://127.0.0.1:8000/brython_dist.js:4174:1
callback http://127.0.0.1:8000/brython_dist.js:4146:1

**Comment by tomamic** [comment]
It happens also for other modules, actually. Names imported through a "star import" statement are only visible inside functions, and not in the main body of the importing script. Does it make sense?

**Comment by jonathanverner** [comment]
I am sorry, I didn't have time to look at it yet, I'll try to find some time this week...

**Comment by PierreQuentel** [comment]
Sorry, I can't reproduce the bug. Running __bounce2d.py__ on my PC starts the application with the sprites moving around and the turtle doing what I tell him ; I have uploaded the application files on brython.info, the [application](http://brython.info/tests/bounce) also works correctly. It uses the latest version (3.3.4).

Maybe the version I use is not the one that causes the bug you observed ?

**Comment by tomamic** [comment]
Yes, sorry, I had to correct the example since I will use it for a course.

But if you import all names, and you move instruction out of the setup function, into the module body directly, the error will appear again. Like here: https://github.com/tomamic/fondinfo/blob/3a7f3c58cd841fab24204f7897a6512245297eae/bounce/bounce_g2d.py

I observed the issue also for other simpler cases: if you import all names from a module, those names will be available in functions, but not in the main script (outside any functions).

Many thanks for helping!
